owasp-top10-2017-apps/a10/games-irados/app/routes.py

```diff
@@ -70,78 +73,128 @@ def decorated_function(*args, **kwargs):
 def root():
     return redirect('/login')
 
+
 @app.route('/logout', methods=['GET'])
 @login_required
 def logout():
     session.clear()
     return redirect('/login')
 
+
 @app.route('/login', methods=['GET', 'POST'])
 def login():
     if request.method == 'POST':
+        log_content = {
+            "ip_adress": request.remote_addr,
+            "username": "unknown",
+            "action": "",
+            "request_route": request.path,
+            "request_method": request.method,
+            "response": "200",
+            "outcome": ""
+        }
         username = request.form.get('username').encode('utf-8')
         psw = Password(request.form.get('password').encode('utf-8'))
         user_password, success = database.get_user_password(username)
+        log_content["action"] = "Login attempt"
         if not success or user_password == None or not psw.validate_password(str(user_password[0])):
-            logger.error("Login error")
+            log_content["outcome"] = "login failed"
+            log_content["response"] = "403"
+            logger.error(log_content)
             flash("Usuario ou senha incorretos", "danger")
             return render_template('login.html')
         session['username'] = username
+        log_content["outcome"] = "login successful"
+        logger.info(log_content)
         return redirect('/home')
     else:
         return render_template('login.html')
 
+
 @app.route('/register', methods=['GET', 'POST'])
 def newuser():
     if request.method == 'POST':
+        log_content = {
+            "ip_adress": request.remote_addr,
+            "username": "unknown",
+            "message": "",
+            "request_route": request.path,
+            "request_method": request.method,
+            "response": "200",
+            "outcome": ""
+        }
         username = request.form.get('username').encode('utf-8')
         psw1 = request.form.get('password1').encode('utf-8')
         psw2 = request.form.get('password2').encode('utf-8')
-
+        log_content["message"] = "Register attempt"
         if psw1 == psw2:
             psw = Password(psw1)
             hashed_psw = psw.get_hashed_password()
             message, success = database.insert_user(username, hashed_psw)
             if success == 1:
-                logger.info("User added")
+                log_content["outcome"] = "Register successful"
+                logger.info(log_content)
                 flash("Novo usuario adicionado!", "primary")
                 return redirect('/login')
             else:
-                logger.error(message)
+                log_content["status"] = "500"
+                log_content["outcome"] = "Register failed"
+                logger.error(log_content)
                 flash(message, "danger")
                 return redirect('/register')
-        
-        logger.error("Passwords don't match")
+        log_content["outcome"] = "Passwords don't match"
+        log_content["status"] = "500"
```

Comment:
I would rather use a `400` (bad request) in this case as the user didn't send both passwords equal and it is not an internal error. What do you think? 😃